packages/core/src/codewhisperer/nextEditPrediction/PredictionTracker.ts

```diff
@@ -0,0 +1,235 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger } from '../../shared/logger/logger'
+import * as diffGenerator from './diffContextGenerator'
+import * as codewhispererClient from '../client/codewhisperer'
+import { predictionTrackerDefaultConfig } from '../models/constants'
+import globals from '../../shared/extensionGlobals'
+
+// defaul values are stored in codewhisperer/model/constants
+export interface FileTrackerConfig {
+    maxFiles: number
+    maxStorageSizeKb: number
+    debounceIntervalMs: number
+    maxAgeMs: number
+    maxSupplementalContext: number
+}
+
+/**
+ * Represents a snapshot of a file at a specific point in time
+ */
+export interface FileSnapshot {
+    filePath: string
+    size: number
+    timestamp: number
+    content: string
+}
+
+export class PredictionTracker {
+    private snapshots: Map<string, FileSnapshot[]> = new Map()
+    private logger = getLogger('nextEditPrediction')
+    readonly config: FileTrackerConfig
+    private storageSize: number = 0
+
+    constructor(extensionContext: vscode.ExtensionContext, config?: Partial<FileTrackerConfig>) {
+        this.config = {
+            ...predictionTrackerDefaultConfig,
+            ...config,
+        }
+    }
+
+    /**
+     * Processes an edit to a document and takes a snapshot if needed
+     * @param document The document being edited
+     * @param previousContent The content of the document before the edit
+     */
+    public async processEdit(document: vscode.TextDocument, previousContent: string): Promise<void> {
```

Comment:
As called out above

Can you please add a global try-catch + error log for all public methods which will be called for completions code path.